core/resources.rs

```diff
@@ -23,9 +25,50 @@ use std::rc::Rc;
 /// Returned by resource read/write/shutdown methods
 pub type AsyncResult<T> = Pin<Box<dyn Future<Output = Result<T, Error>>>>;
 
-/// All objects that can be store in the resource table should implement the
-/// `Resource` trait.
-/// TODO(@AaronO): investigate avoiding alloc on read/write/shutdown
+/// Resources are Rust objects that are attached to a [deno_core::JsRuntime].
+/// They live at most as long as the runtime lives. They are identified in JS
+/// by a numeric ID (the resource ID, or rid). Resources can be created in ops.
+/// Resources can also be retrieved in ops by their rid. Resources are not
+/// thread-safe - they can only be accessed from the thread that the JsRuntime
+/// lives on.
+///
+/// Resources are reference counted in Rust. This means that they can be
+/// cloned and passed around. When the last reference is dropped, the resource
+/// is automatically closed. As long as the resource exists in the resource
+/// table, the reference count is at least 1.
+///
+/// ### Readable
+///
+/// Readable resources are resources that can have data read from. Examples of
+/// this are files, sockets, or HTTP responses.
```

Comment:
Shouldn't it be HTTP requests?

Reply:
for `fetch` it's responses, for the server it is requests. I changed it to "HTTP streams".